### PR TITLE
Remove the "project" concept

### PR DESCRIPTION
**Pull request by jsnajdr**
The sections list and the boot functions are divided into "common" ones and into "project-specific" ones, where the only project we ever had is `wordpress-com`.

That separation didn't prove useful for anything and this PR merges everything together.

There is only one `sections.js` module now, and all the boot functions have been merged to `client/app/boot/common.js`.

The `lib/path-to-section` helper has been updated to use the `sections-helper` module rather than importing `sections` directly. That should simplify the dependency graph and make webpack's job easier. See the comments inside `sections-helper` to learn more about the motivation.

### CODE REVIEW

**Review by blowery — APPROVED** [review]
All this looks good to me, though it would be nice to figure out why the e2e's and the canary is failing.

**Comment by jsnajdr** [comment]
> All this looks good to me, though it would be nice to figure out why the e2e's and the canary is failing.

It seems to be a calypso.live thing. I'm investigating and will ensure it's green before merging.

I resolved some conflicts with @belcherj's #32414 that moves `/sharing` to `/marketing`. I opted for a different solution than the original PR. Instead of two sections, I define only one with two paths. Please review 979d53b.

**Comment by jsnajdr** [comment]
All e2e's are all green 💚 I'll wait for @belcherj's feedback on the marketing section and then merge.

**Pull request merged**
